android/src/main/java/com/mattermost/networkclient/Helper.kt

```diff
@@ -17,10 +26,65 @@ fun Response.returnAsWriteableMap (): WritableMap {
     return map;
 }
 
-fun Response.handleViaPromise(promise: Promise){
-    return if(this.isSuccessful){
+/**
+ * Completes the response handling by calling a promise resolve / reject based on response type
+ *
+ * @param promise The promise to resolve/reject
+ */
+fun Response.promiseResolution(promise: Promise): Response {
+    if (this.isSuccessful) {
         promise.resolve(this.returnAsWriteableMap());
     } else {
         promise.reject(this.code.toString(), this.returnAsWriteableMap())
     }
+    return this
+}
+
+/**
+ * Parses options passed in over the bridge for individual requests
+ *
+ * @param options ReadableMap of options from the App
+ */
+fun Request.Builder.parseOptions(options: ReadableMap): Request.Builder {
+    return this
```

Comment:
todo ?

Reply:
Almost! It's used in https://github.com/mattermost/react-native-network-client/pull/19 - just rebase conflicts led me to adding this here / now, so I can use it a few PR's later.